lib/manager/pipenv/update.js

```diff
@@ -0,0 +1,19 @@
+toml = require('toml')
+tomlify = require('tomlify-j0.4')
+
+module.exports = {
+  updateDependency,
+};
+
+function updateDependency(fileContent, upgrade) {
+  try {
+    logger.debug(`pipenv.updateDependency(): ${upgrade.newValue}`);
+    const pipfile = toml.parse(fileContent);
+    pipfile.packages[upgrade.depName] = upgrade.newValue;
+    return tomlify.toToml(pipfile)
```

Comment:
How consistent is TOML formatting, including whitespace? Is this approach likely to produce a large diff the first time Renovate updates any dependency, if the output for `tomlify.toToml()` doesn't match with the user's previous formatting?

Reply:
I have no idea. I only learned that the files are toml files and not ini files today. 

I tested it manually and it managed to only change the one line that needed changing (plus removing the trailing newline, which I can fix). It’s just one experiment with only a simple Pipfile, but it seemed promising.

Reply:
That’s a good sign. Usually I go for a “keyhole surgery” approach to changing files rather than full parse/stringify because people can often be very particular about the way they want to format their files

Reply:
Yeah, I'm nervous about the parse and stringify strategy too. On the other hand, I'm not super confident that I can implement a good enough ad hoc parser to enable the keyhole strategy.

What I really want is a toml parser that gives me the absolute line number. Then I could get a reliable parse of the whole file and only change the one part I care about.

Failing that, I'm going to try and add some test coverage based on more complex Pipfiles.